### PR TITLE
internal: Drop support for v2 xDS resource version

### PR DESCRIPTION
**Pull request by stevesloka**
Drop support for xDS v3 resource version by changing the default
boostrap resource version to v3 and removing the v2 xDS resource server.

Updates #1898

Signed-off-by: Steve Sloka <slokas@vmware.com>

### CODE REVIEW

**Comment by skriss** [comment]
We should update https://github.com/projectcontour/contour/blob/main/cmd/contour/bootstrap.go#L37 and the associated documentation to default to `v3` as well.

**Review by youngnick — COMMENTED** [review]
+5, -1395, impressive!

LGTM.

**Comment by stevesloka** [comment]
@skriss I updated the default bootstrap value, thanks, I missed that. Also I removed the flag from the initContainer since it's not needed anymore.

File: cmd/contour/bootstrap.go [hunk 1]
```diff
@@ -34,6 +34,6 @@ func registerBootstrap(app *kingpin.Application) (*kingpin.CmdClause, *envoy.Boo
 	bootstrap.Flag("envoy-cert-file", "Client certificate filename for Envoy secure xDS gRPC communication.").Envar("ENVOY_CERT_FILE").StringVar(&config.GrpcClientCert)
 	bootstrap.Flag("envoy-key-file", "Client key filename for Envoy secure xDS gRPC communication.").Envar("ENVOY_KEY_FILE").StringVar(&config.GrpcClientKey)
 	bootstrap.Flag("namespace", "The namespace the Envoy container will run in.").Envar("CONTOUR_NAMESPACE").Default("projectcontour").StringVar(&config.Namespace)
-	bootstrap.Flag("xds-resource-version", "The versions of the xDS resources to request from Contour.").Default("v2").StringVar((*string)(&config.XDSResourceVersion))
+	bootstrap.Flag("xds-resource-version", "The versions of the xDS resources to request from Contour.").Default("v3").StringVar((*string)(&config.XDSResourceVersion))
```
Comment (skriss) [inline cmd/contour/bootstrap.go:37]: so I guess we're now ignoring this flag and just automatically serving v3 only, right? I guess that's fine, maybe worth a callout in docs or help text though.

**Review by skriss — APPROVED** [review]
Need to update the docs for the flag too: https://github.com/projectcontour/contour/blob/main/site/docs/main/configuration.md#bootstrap-config-file

Otherwise LGTM.

File: cmd/contour/bootstrap.go (hunk 1 above)
Comment (stevesloka) [reply cmd/contour/bootstrap.go:37]: Yeah essentially the flag does nothing now until we need to support say v4.

Comment (jpeach) [reply cmd/contour/bootstrap.go:37]: I'm way late to this party, but I agree that keeping the flag makes sense. Do you think that emitting an error for `--xds-resource-version=v2` is worth consideration?

**Pull request merged**